src/Bridge/Graphql/Resolver/CollectionResolverFactory.php

```diff
@@ -0,0 +1,109 @@
+ */
+final class CollectionResolverFactory extends AbstractResolverFactory implements CollectionResolverFactoryInterface
+{
+    private $collectionDataProvider;
+    private $normalizer;
+    private $identifiersExtractor;
+    private $requestStack;
+    private $paginationEnabled;
+
+    public function __construct(CollectionDataProviderInterface $collectionDataProvider, SubresourceDataProviderInterface $subresourceDataProvider, NormalizerInterface $normalizer, IdentifiersExtractorInterface $identifiersExtractor, RequestStack $requestStack = null, bool $paginationEnabled = false)
+    {
+        parent::__construct($subresourceDataProvider);
+
+        $this->collectionDataProvider = $collectionDataProvider;
+        $this->normalizer = $normalizer;
+        $this->identifiersExtractor = $identifiersExtractor;
+        $this->requestStack = $requestStack;
+        $this->paginationEnabled = $paginationEnabled;
+    }
+
+    /**
+     * @throws \InvalidArgumentException
+     */
+    public function createCollectionResolver(string $resourceClass, string $rootClass): callable
+    {
+        return function ($root, $args, $context, ResolveInfo $info) use ($resourceClass, $rootClass) {
+            $request = $this->requestStack ? $this->requestStack->getCurrentRequest() : null;
+            if (null !== $request) {
+                $request->attributes->set(
+                    '_graphql_collections_args',
+                    [$resourceClass => $args] + $request->attributes->get('_graphql_collections_args', [])
+                );
+            }
+
+            $rootProperty = $info->fieldName;
+            if (isset($root[$rootProperty])) {
+                $rootIdentifiers = $this->identifiersExtractor->getIdentifiersFromResourceClass($rootClass);
+                $subresource = $this->getSubresource($rootClass, $root, $rootIdentifiers, $rootProperty, $resourceClass, true);
+                $collection = $subresource ?? [];
+            } else {
+                $collection = $this->collectionDataProvider->getCollection($resourceClass);
+            }
+
+            if (!$this->paginationEnabled) {
+                $data = [];
+                foreach ($collection as $index => $object) {
+                    $data[$index] = $this->normalizer->normalize($object, null, ['graphql' => true]);
+                }
+
+                return $data;
+            }
+
+            $offset = 0;
+            if (isset($args['after'])) {
+                $after = \base64_decode($args['after'], true);
+                if (false === $after) {
+                    throw new InvalidArgumentException(sprintf('Cursor %s is invalid', $args['after']));
+                }
+                $offset = (int) $after;
+                ++$offset;
+            }
+
+            $data = ['edges' => [], 'pageInfo' => ['endCursor' => null, 'hasNextPage' => false]];
+            if ($collection instanceof PaginatorInterface && ($totalItems = $collection->getTotalItems()) > 0) {
```

Comment:
Can't we also support `PartialPaginatorInterface`?

Reply:
👍 but in another PR please